e2e/nomostest/gitproviders/repository.go

```diff
@@ -581,9 +579,21 @@ func (g *Repository) Push(args ...string) error {
 	return nil
 }
 
+func (g *Repository) push(remoteURL string, args ...string) error {
```

Comment:
Why pull out an inner push method?
Don't we want to retry when calling PushAllBranches?

Reply:
This is so we can pass the `remote` with `PushAllBranches` in order to use the `port` value passed into the callback instead of calling the `LocalPort` method. This enables us to wrap the callback in the lock without causing a deadlock.

In this case the `RemoteURL` is static, so we don't retry for the `RemoteURL`. I could refactor this to retry for failures in `git push`, but I opted for just pushing without a retry